lib/message.rb

```diff
@@ -0,0 +1,70 @@
+require 'pg'
+
+class Message
+  attr_reader :id, :message, :date
+
+  def initialize(id:, message:, date:)
+    @id = id
+    @message = message
+    @date = date
+  end
+
+  
+  def self.filter(keyword:) 
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+    result = connection.exec("SELECT * FROM peeps")
+    result.map do |message|
+      if message['message'].include?(keyword)
+        Message.new(id: message['id'], message: message['message'], date: message['_date'])
+      else
+        Message.new(id: nil, message: 'Not a  Match', date: nil)
+      end
+    end
+           
+          
+         
+  end
+
+  def self.sort
```

Comment:
You have a bug in this method, which I only caught because I ran your app and could see in the browser the sort functionality wasn't working. Your tests were giving you a false positive, which leads me to believe you aren't in the habit of checking your work by actually running it in the browser.

If this is so, please be careful here. Tests are very valuable, but even they can lead you astray if they are set up incorrectly. A web developer should always check their software as if they were the user, in addition to running automated tests. This will help you not only catch more bugs, but help you find UX issues you might otherwise miss by just looking at your code.